src/RouteCollector.php

```diff
@@ -25,6 +32,12 @@ public function addRoute(Route|Group ...$routes): RouteCollectorInterface
         return $this;
     }
 
+    public function addResource(ResourceInterface $resource): RouteCollectorInterface
```

Comment:
It need to add resources in runtime? If no, better pass resources array to constructor.

Reply:
Yes, it will be added in runtime like routes.